packages/grid/src/Grid.tsx

```diff
@@ -1907,6 +1907,19 @@ class Grid extends PureComponent<GridProps, GridState> {
       metrics.rowHeight
     );
 
+    // Check if at the top and attempting to scroll up
+    // Or at the bottom and attempting to scroll down
+    // Or at the left and attempting to scroll left
+    // Or at the right and attempting to scroll right
+    if (
```

Comment:
I think you need something that only returns if neither vertical or horizontal movement will result from the delta.

e.g. I haven't tested but something like:
```typescript
const willScrollX = left > 0 && deltaX < 0 || left < lastLeft && delta > 0
const willScrollY = top > 0 && deltaY < 0 || top < lastTop && delta > 0

if (!willScrollX && !willScrollY) {
  return
}
```

Reply:
It is behaving the same as my commit, but makes more sense semantically, pushed change.